### DISCUSSION TITLE
Solve inconsistencies between "paused" mutations and "paused" queries

### DISCUSSION THREAD

**Discussion by crutchcorn** [TanStack/query#6760] [Ideas]
While working on adding in a banner to display offline metadata to our users, I noticed that there was no easy way to fetch the `paused` mutations count, even while using the `useMutationState` hook.

This seems to be due to the fact that mutations do not have a `paused` status enum, despite having a `isPaused` state property.

This seems to not align well with how queries are handled. **It might be nice to have a `paused` status enum on mutations as they are on queries**.

# References to `"paused"` mutations:

https://github.com/TanStack/query/blob/main/packages/query-core/src/mutation.ts#L35
https://github.com/TanStack/query/blob/main/packages/query-core/src/mutation.ts#L63-L65

# Where I'd want to add support for `paused` status type:

- https://github.com/TanStack/query/blob/main/packages/query-core/src/mutationObserver.ts#L135
    - Add a `isPaused` derived state, like there is for `useQuery`
- https://github.com/TanStack/query/blob/main/packages/query-core/src/mutation.ts#L288
    - Add `status: 'paused'` to the `...state` spread
- https://github.com/TanStack/query/blob/main/packages/query-core/src/types.ts#L686
    - Worth mentioning that [`QueryStatus` is broken into `FetchStatus` as well](https://github.com/TanStack/query/blob/main/packages/query-core/src/types.ts#L452-L453). Maybe we want to do the same for mutations?

**Comment by TkDodo** [comment]
> I noticed that there was no easy way to fetch the paused mutations count

Yeah, queries and mutations have diverged a bit. There is no `fetchStatus` because with mutations, we can and want to have an `idle` primary state.

What we could do is add an `isPaused: boolean` filter to `MutationFilters`. What you can do right now is to use:

```
predicate: mutation => mutation.state.isPaused
```

`MutationFilters` can be passed to e.g. `useMutationState`. We do this internally, too:


https://github.com/TanStack/query/blob/5290aa10121e512914ee888512a36f9a05820278/packages/query-core/src/mutationCache.ts#L166

**Reply by crutchcorn** [reply]
For sure! I am now using `isPaused` in my codebase, so this isn't a blocker for today.

> Yeah, queries and mutations have diverged a bit.

Is this something we may want to investigate for a potential Query v6 release? I'm happy to take on this work
